components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/geo/service/GeoLocationProviderServiceTest.java

```diff
@@ -0,0 +1,255 @@
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.device.mgt.common.Device;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.DeviceManagementConstants;
+import org.wso2.carbon.device.mgt.common.geo.service.Alert;
+import org.wso2.carbon.device.mgt.common.geo.service.GeoFence;
+import org.wso2.carbon.device.mgt.core.TestDeviceManagementService;
+import org.wso2.carbon.device.mgt.core.authorization.DeviceAccessAuthorizationServiceImpl;
+import org.wso2.carbon.device.mgt.core.common.TestDataHolder;
+import org.wso2.carbon.device.mgt.core.internal.DeviceManagementDataHolder;
+import org.wso2.carbon.device.mgt.core.internal.DeviceManagementServiceComponent;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderService;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderServiceImpl;
+import org.wso2.carbon.device.mgt.core.service.GroupManagementProviderServiceImpl;
+import org.wso2.carbon.event.processor.stub.EventProcessorAdminServiceStub;
+import org.wso2.carbon.identity.jwt.client.extension.exception.JWTClientException;
+import org.wso2.carbon.registry.core.config.RegistryContext;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.internal.RegistryDataHolder;
+import org.wso2.carbon.registry.core.jdbc.realm.InMemoryRealmService;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.io.InputStream;
+import java.rmi.RemoteException;
+import java.util.List;
+
+import static org.powermock.api.mockito.PowerMockito.*;
+
+@PrepareForTest(GeoLocationProviderServiceImpl.class)
+public class GeoLocationProviderServiceTest {
+
+    private static final Log log = LogFactory.getLog(GeoLocationProviderServiceTest.class);
+    private static final String DEVICE_TYPE = "GL_TEST_TYPE";
+    private static final String DEVICE_ID = "GL-TEST-DEVICE-ID-1";
+    private static final String SAMPLE_GEO_JSON = "12121";
+    private static final String SAMPLE_AREA_NAME = "CUSTOM_NAME";
+    private static final String SAMPLE_QUERY_NAME = "QUERY_NAME";
+    private static final String SAMPLE_PROXIMITY_DISATANCE = "100";
+    private static final String SAMPLE_PROXIMITY_TIME = "50";
+    private static final String SAMPLE_SPEED_ALERT_VALUE = "120";
+    private static final String SAMPLE_STATIONARY_TIME = "1500";
+    private static final String SAMPLE_FLUCTUATION_RADIUS = "2000";
+
+    private EventProcessorAdminServiceStub mockEventProcessorAdminServiceStub;
+    private GeoLocationProviderServiceImpl geoLocationProviderServiceImpl;
+
+    @BeforeClass
+    public void init() throws Exception {
+        initMocks();
+        enrollTestDevice();
+    }
+
+    @Test
```

Comment:
Pleaase add the description for the test. Please change it in other places as well.